src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -37,37 +39,77 @@
 
   ParameterizedTypeName(ClassName rawType, List<TypeName> typeArguments,
       List<AnnotationSpec> annotations) {
+    this(rawType, null, typeArguments, annotations);
+  }
+
+  private ParameterizedTypeName(ClassName rawType, ParameterizedTypeName enclosingType,
+      List<TypeName> typeArguments, List<AnnotationSpec> annotations) {
     super(annotations);
     this.rawType = checkNotNull(rawType, "rawType == null");
+    this.enclosingType = enclosingType;
     this.typeArguments = Util.immutableList(typeArguments);
 
-    checkArgument(!this.typeArguments.isEmpty(), "no type arguments: %s", rawType);
+    checkArgument(
+        !this.typeArguments.isEmpty() || enclosingType instanceof ParameterizedTypeName,
+        "no type arguments: %s",
+        rawType);
     for (TypeName typeArgument : this.typeArguments) {
       checkArgument(!typeArgument.isPrimitive() && typeArgument != VOID,
           "invalid type parameter: %s", typeArgument);
     }
   }
 
   @Override public ParameterizedTypeName annotated(List<AnnotationSpec> annotations) {
-    return new ParameterizedTypeName(rawType, typeArguments, concatAnnotations(annotations));
+    return new ParameterizedTypeName(
+        rawType, enclosingType, typeArguments, concatAnnotations(annotations));
   }
 
   @Override public TypeName withoutAnnotations() {
-    return new ParameterizedTypeName(rawType, typeArguments);
+    return new ParameterizedTypeName(
+        rawType, enclosingType, typeArguments, new ArrayList<AnnotationSpec>());
   }
 
   @Override CodeWriter emit(CodeWriter out) throws IOException {
-    rawType.emitAnnotations(out);
-    rawType.emit(out);
-    out.emitAndIndent("<");
-    boolean firstParameter = true;
-    for (TypeName parameter : typeArguments) {
-      if (!firstParameter) out.emitAndIndent(", ");
-      parameter.emitAnnotations(out);
-      parameter.emit(out);
-      firstParameter = false;
+    if (enclosingType != null) {
+      enclosingType.emitAnnotations(out);
```

Comment:
Do we always want to get annotations from `rawType` ? That seems like it’s least likely to surprise?